include/zephyr/sys/list_gen.h

```diff
@@ -234,4 +234,15 @@
 		return false;						 \
 	}
 
+#define Z_GENLIST_LEN(__lname, __nname)                                                            \
+	static inline size_t sys_##__lname##_len(sys_##__lname##_t * list)                         \
+	{                                                                                          \
+		size_t len = 0;                                                                    \
+		static sys_##__nname##_t * node;                                                   \
```

Comment:
I hope it's okay to comment even though the PR has already been merged for a long time.  

I have a question: why is the iterator variable `node` defined as `static` linkage?

Reply:
Good question. @OneTone - if you make a PR to fix it, I would approve

Reply:
Thank you @cfriedt !  
I've just created a PR to address the issue: #76973 .  
It's my first contribution to this project, so please feel free to point out any problems.  
I'd really appreciate your feedback!